frontend/src/qri/ref.ts

```diff
@@ -150,6 +152,17 @@ export function humanRef(ref: QriRef): QriRef {
   return newQriRef({ username : ref.username, name: ref.name })
 }
 
+// downloadLinkFromQriRef creates a download link
+export function downloadLinkFromQriRef(ref: QriRef, body: boolean = false): string {
+  console.log('theRef', ref)
+  let pathSegment = ref.path ? `/at${ref.path}` : ''
+
+  if (body) {
+    return `${API_BASE_URL}/ds/get/${ref.username}/${ref.name}${pathSegment}/body.csv`
+  }
+  return `${API_BASE_URL}/download/${ref.username}/${ref.name}${pathSegment}`
+}
```

Comment:
Move this to `features/dataset/state/datasetActions.ts` instead.

The qri dir should be treated like an NPM module that cannot import anything outside it's own folder. This function relies on our application-specific URL structure